app/modules/platform/saga/task/signinSSO.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import { type Saga } from 'redux-saga';
+import { put } from 'redux-saga/effects';
+
+import users from 'modules/users';
+
+import actions from '../../actions';
+import * as a from '../../actionTypes';
+import * as m from '../../model';
+
+const signinSSO = function* (action: a.SigninSSOAction): Saga<void> {
+  const { apiToken, userId } = action.payload;
+
+  // Dispatch action to retrieve the current user
+  yield put(users.actions.fetch(userId));
```

Comment:
Fetching the user is not necessary in this saga; it happens automatically when the user object is needed (for example when the navbar is rendered).